tests/sha_ut/sha_ut.c

```diff
@@ -0,0 +1,450 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#else
+#include <stdlib.h>
+#endif
+
+/**
+ * Include the C standards here.
+ */
+#ifdef __cplusplus
+#include <cstddef>
+#include <ctime>
+#else
+#include <stddef.h>
+#include <time.h>
+#endif
+
+/**
+ * Include the test tools.
+ */
+#include "testrunnerswitcher.h"
+#include "umock_c.h"
+#include "umocktypes_charptr.h"
+#include "umock_c_negative_tests.h"
+#include "azure_c_shared_utility/macro_utils.h"
+
+#define ENABLE_MOCKS
+#undef ENABLE_MOCKS
+
+#include "azure_c_shared_utility/sha.h"
+
+DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+{
+    ASSERT_FAIL("umock_c reported error :%s", ENUM_TO_STRING(UMOCK_C_ERROR_CODE, error_code));
+}
+
+/**
+ * This is necessary for the test suite, just keep as is.
+ */
+static TEST_MUTEX_HANDLE g_testByTest;
+
+BEGIN_TEST_SUITE(sha_ut)
+
+    /**
+     * This is the place where we initialize the test system. Replace the test name to associate the test
+     *   suite with your test cases.
+     * It is called once, before start the tests.
+     */
+    TEST_SUITE_INITIALIZE(TestClassInitialize)
+    {
+        g_testByTest = TEST_MUTEX_CREATE();
+        ASSERT_IS_NOT_NULL(g_testByTest);
+
+        (void)umock_c_init(on_umock_c_error);
+    }
+
+    /**
+     * The test suite will call this function to cleanup your machine.
+     * It is called only once, after all tests is done.
+     */
+    TEST_SUITE_CLEANUP(TestClassCleanup)
+    {
+        umock_c_deinit();
+
+        TEST_MUTEX_DESTROY(g_testByTest);
+    }
+
+    /**
+     * The test suite will call this function to prepare the machine for the new test.
+     * It is called before execute each test.
+     */
+    TEST_FUNCTION_INITIALIZE(initialize)
+    {
+        if (TEST_MUTEX_ACQUIRE(g_testByTest))
+        {
+            ASSERT_FAIL("Could not acquire test serialization mutex.");
+        }
+
+        umock_c_reset_all_calls();
+    }
+
+    TEST_FUNCTION_CLEANUP(cleans)
+    {
+        TEST_MUTEX_RELEASE(g_testByTest);
+    }
+
+    TEST_FUNCTION(SHA224Reset_succeed)
```

Comment:
>succeed [](start = 30, length = 7)

succeeds

Reply:
For the other tests too

---
In reply to: [237965997](https://github.com/Azure/azure-c-shared-utility/pull/236#discussion_r237965997) [](ancestors = 237965997)